[jon-seana] iP

jon-seana opened the pull request:
# **Botzilla 😃**

### "Software is a great combination of artistry and engineering." - Bill Gates ([source](https://blog.tmetric.com/best-short-quotes-for-motivation-for-software-developers/))

Botzilla frees your mind and thoughts of having to remember things you need to do. It's,

- text-based
- ~~easy to learn~~ and understand
- _SUPER FAST_ and **_EFFICIENT_** to use

All you need to do is,

1. download it from [here](https://github.com/jon-seana/ip/releases/tag/A-Jar).
2. double-click on botzilla.jar file.
3. add your tasks by following the command format specified.
4. let it manage your tasks for you :)

And it is COMPLETELY FREE!

Features:

- [x] Managing tasks
- [x] Managing deadlines (coming soon)
- [x] Reminders (coming soon)
- [ ] Alphabetical ordering of task (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method:
```java
public static void main(String[] args) {
        new Botzilla().run();
 }

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jon-seana committed:
add Botzilla.java

jon-seana committed:
Added Echo functionality

jon-seana committed:
Added Add and List functionalities

jon-seana committed:
Added mark as done functionality

jon-seana committed:
Added ToDos, Events, Deadlines functionalities

jon-seana committed:
Fix ToDos, Events, Deadlines

jon-seana committed:
Did automated text UI testing

jon-seana committed:
add Handle Errors functionalities

jon-seana committed:
Add delete functionality, minor bug fixes and improvements to the code

jon-seana committed:
made save function

jon-seana committed:
add dates and times function

jon-seana committed:
Merge branch 'branch-Level-8'

jon-seana committed:
Add more OOP

jon-seana committed:
Merge branch 'branch-A-MoreOOP'

jon-seana committed:
Add packages

jon-seana committed:
Merge branch 'branch-A-Packages'

jon-seana committed:
Merge branch 'add-gradle-support'

jon-seana committed:
Add JUnit tests

jon-seana committed:
Merge branch 'branch-A-JUnit'

jon-seana committed:
Create JAR file

jon-seana committed:
Add JavaDoc comments

jon-seana committed:
Fix JavaDoc errors

jon-seana committed:
Fix coding standard

jon-seana committed:
Add find function

jon-seana committed:
Merge branch 'branch-A-JavaDoc'

jon-seana committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/botzilla/task/TaskList.java

jon-seana committed:
Merge branch 'branch-Level-9'

jon-seana committed:
Fix minor bugs

ZaydM18 commented on src/main/java/botzilla/command/Parser.java:
```diff
@@ -0,0 +1,115 @@
+                System.out.println(endFormat);
+            } else if (input.startsWith("find")) {
+                if (input.length() <= 5) {
+                    throw new BotzillaException(horizontalLine + "\n"
```
Good use of spacing to add clarity in a long line of code.

ZaydM18 commented on src/main/java/botzilla/task/Event.java:
```diff
@@ -0,0 +1,121 @@
+    @Override
+    public String saveData() {
+        if (fromDate != null && toDate != null) {
+            return "[E]" + super.toString() + " (from: " + fromDate.format(DateTimeFormatter.ofPattern("MMM d yyyy, h:mm a")) + " to: " + toDate.format(DateTimeFormatter.ofPattern("MMM d yyyy, h:mm a")) + ")";
```
Though most other code blocks have very clear spacing, a few methods like saveData() have long lines of code that could be separated to be clearer. Very minor issue.

ZaydM18 commented on src/main/java/botzilla/Botzilla.java:
```diff
@@ -0,0 +1,52 @@
+package botzilla;
+
+import botzilla.command.Parser;
```
General consistency in import statement declarations throughout the code. Good to see.

ZaydM18 commented on src/main/java/botzilla/task/Deadline.java:
```diff
@@ -0,0 +1,103 @@
+     * @param ui Ui.
+     * @return Deadline Deadline.
+     */
+    public static Deadline createDeadline(String input, Ui ui) {
```
You could leave more line gaps in the createDeadline method, since it's a long and winding one. Minor problem.

ZaydM18 reviewed (commented):
Overall, well-written and follows the coding standards for the most part, with only very minor issues in line breaks and spacing.

jiangsuwangjing commented on src/main/java/botzilla/command/Parser.java:
```diff
@@ -0,0 +1,115 @@
+import botzilla.task.Todo;
+import botzilla.task.Deadline;
+import botzilla.task.Event;
+import botzilla.exception.BotzillaException;
```
This may not pass the style check because it does not follow alphabetical order.

jiangsuwangjing commented on src/main/java/botzilla/storage/Storage.java:
```diff
@@ -0,0 +1,105 @@
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
```
It is better to put Java packages before self-created packages.

jiangsuwangjing commented on src/main/java/botzilla/ui/Ui.java:
```diff
@@ -0,0 +1,158 @@
+    /**
+     * Method to print out the delete warning message.
+     */
+    public void deleteError() {
```
For errors, it is clearer to name it as show***Error(). Other wise it might sound like throwing a new error.

jiangsuwangjing commented on src/test/java/botzilla/command/DeadlineTest.java:
```diff
@@ -0,0 +1,102 @@
+     * Test creating a Deadline with valid input using the dash date format.
+     */
+    @Test
+    public void createDeadline_validInput_dashFormat_success() {
```
Good naming convention for test cases!

jiangsuwangjing commented on src/main/java/botzilla/task/Task.java:
```diff
@@ -0,0 +1,60 @@
+    /**
+     * Method to make isDone = true.
+     */
+    public void markAsDone() {
```
Consistent function names, good job!

jiangsuwangjing reviewed (commented):
Overall very good coding standard compliance, well done!

jon-seana committed:
Add checkstyle function

jon-seana committed:
Merge branch 'branch-A-CheckStyle'

jon-seana committed:
Add GUI function

jon-seana committed:
Merge branch 'branch-Level-10'

jon-seana committed:
Fix code standards

jon-seana committed:
Minor changes in build.gradle file

jon-seana committed:
Add Assertions to the code

There are several classes where assertions have been added to catch
potential programming errors during development.

Add Assertions to the code in classes: Storage, Deadline, Event,
TaskList, Todo, DialogBox, Main, MainWindow.

These assertions help to ensure input validity and detect
unexpected states early.

jon-seana committed:
Refactor code for improved quality and maintainability

Several classes were refactored to enhance code clarity, reduce the
amount of redundancy, and improve the overall code maintainability.

Classes which have been refactored are:Botzilla.java, Main.java,
MainWindow.java, Parser.java, Storage.java, Deadline.java,
Event.java, TaskList.java, Todo.java, Ui.java, DeadlineTest.java,
EventTest.java

These changes ensure an improvement in code quality and easier
readability for future developers.

jon-seana committed:
Fix minor issues with JavaDoc header comments

jon-seana committed:
Merge pull request #2 from jon-seana/branch-A-Assertions

Add Assertions to the code. Verified to have no conflicts.

jon-seana committed:
Resolve merge conflicts

jon-seana committed:
Merge pull request #3 from jon-seana/branch-A-CodeQuality

Branch a code quality. Merge conflicts have been resolved.

jon-seana committed:
Add gui package to contain various elements of the FXML classes

jon-seana committed:
Merge branch 'branch-package'

jon-seana committed:
Create Botzilla.yml

Added GitHub Actions for CI workflow

jon-seana committed:
Add GitHub Actions functionality

jon-seana committed:
Add sorting for deadline and event tasks

With this change, user is now able to input the command "sort",
which will sort in ascending order all of the deadline and event
tasks.

A sortTaskList() method is implemented in TaskList class, to allow
for this feature to work in alignment with the OOP guidelines.

This change improves task management by allowing users to see
their most urgent and time sensitive tasks at a glance.

jon-seana committed:
Merge branch 'branch-C-Sort'

jon-seana committed:
Improve GUI, personality and error handling

To enhance the chatbot's graphical user interface, for a more engaging
and pleasant user experience by specifying the remedies for each errors
in a unique and tailored solution.

Change some areas of the code to include:
- Better layout and spacing of GUI elements.
- Update colors, fonts and icons of GUI.
- Add more tailored responses to user's inputs.
- Add more checks to handle common formatting errors.
- Enhance chatbot response to be more human like.

These improvements provide a modern, friendly and more immersive
interface that reflects the chatbot's unique character.

jon-seana committed:
Merge branch 'branch-A-BetterGui'

jon-seana committed:
Add user guide and fix code format

jon-seana committed:
Merge branch 'branch-A-MoreTesting'

jon-seana committed:
Fix readme format

jon-seana committed:
Update README.md

jon-seana committed:
Update README.md

jon-seana committed:
update readme format

jon-seana committed:
Fix FXML version issue

jon-seana committed:
Fix minor code format issue

jon-seana committed:
Fix code abstraction issue

jon-seana committed:
Fix Parser.java bug

jon-seana committed:
Add new README.md contents
